--- origsrc/libepoxy-1.2/configure.ac	2015-05-17 11:18:26.664510600 -0500
+++ src/libepoxy-1.2/configure.ac	2015-05-17 10:19:03.389032100 -0500
@@ -122,7 +122,7 @@ AC_CHECK_LIB([dl], [dlopen], [DLOPEN_LIB
 AC_SUBST([DLOPEN_LIBS])
 
 case $host_os in
-    mingw*)
+    cygwin*|mingw*)
         # visibility flags aren't supported for windows DLLs, and the
         # compiler whines to tell you so, so don't set them up.
         ;;
