--- origsrc/libepoxy-1.4.3/src/dispatch_common.c	2017-06-06 04:55:43.000000000 -0500
+++ src/libepoxy-1.4.3/src/dispatch_common.c	2017-09-04 16:21:21.088483700 -0500
@@ -177,6 +177,8 @@
 #define GLX_LIB "/opt/X11/lib/libGL.1.dylib"
 #elif defined(ANDROID)
 #define GLX_LIB "libGLESv2.so"
+#elif defined(__CYGWIN__)
+#define GLX_LIB "cygGL-1.dll"
 #else
 #define GLX_LIB "libGL.so.1"
 #endif
@@ -185,6 +187,10 @@
 #define EGL_LIB "libEGL.so"
 #define GLES1_LIB "libGLESv1_CM.so"
 #define GLES2_LIB "libGLESv2.so"
+#elif defined (__CYGWIN__)
+#define EGL_LIB "cygEGL-1.dll"
+#define GLES1_LIB "cygGLESv1_CM-1.dll"
+#define GLES2_LIB "cygGLESv2-2.dll"
 #elif defined _WIN32
 #define EGL_LIB "libEGL.dll"
 #define GLES1_LIB "libGLES_CM.dll"
